apps/remix-ide/src/remixAppManager.js

```diff
@@ -149,6 +149,16 @@ export class RemixAppManager extends PluginManager {
       pattern: [],
       sticky: true
     })
+    await this.call('filePanel', 'registerContextMenuItem', {
```

Comment:
@HDauven note that for this to work, the plugin side should implement the `compileCustomAction` function, is this the case?
cc @bunsenstraat

Reply:
We can merge the PR now, but `compileCustomAction` must be added in the plugin codebase.

Reply:
please reach out in our channel https://gitter.im/ethereum/remix-dev if you want more information about that.

Reply:
Hey @yann300! 

The plugin actually does implement this. There is a hook that extends the PluginClient provided by Remix. This RemixClient adds both a setCompilerCallback callback function, compileCustomAction function and a compile function.

Reply:
Note that I've tested the above mentioned functionality extensively, both regular compilations and compilations through the context menu, both locally and by plugging in the locally ran plugin into the hosted remix environment.